src/Microsoft.OpenApi/Models/References/OpenApiExampleReference.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. 
+
+using System;
+using System.Collections.Generic;
+using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.Interfaces;
+using Microsoft.OpenApi.Properties;
+using Microsoft.OpenApi.Writers;
+
+namespace Microsoft.OpenApi.Models.References
+{
+    /// <summary>
+    /// Example Object Reference.
+    /// </summary>
+    internal class OpenApiExampleReference : OpenApiExample
+    {
+        private OpenApiExample _target;
+        private readonly OpenApiReference _reference;
+        private string _summary;
+        private string _description;
+
+        private OpenApiExample Target
+        {
+            get
+            {
+                _target ??= _reference.HostDocument.ResolveReferenceTo<OpenApiExample>(_reference);
+                return _target;
+            }
+        }
+
+        /// <summary>
+        /// Constructor initializing the reference object.
+        /// </summary>
+        /// <param name="referenceId">The reference Id.</param>
+        /// <param name="hostDocument">The host OpenAPI document.</param>
+        /// <param name="externalResource">Optional: External resource in the reference.
+        /// It may be:
+        /// 1. a absolute/relative file path, for example:  ../commons/pet.json
+        /// 2. a Url, for example: http://localhost/pet.json
+        /// </param>
+        public OpenApiExampleReference(string referenceId, OpenApiDocument hostDocument, string externalResource = null)
+        {
+            if (string.IsNullOrEmpty(referenceId))
+            {
+                throw Error.Argument(nameof(referenceId), SRResource.ReferenceIdIsNullOrEmpty);
+            }
+            if (hostDocument == null)
+            {
+                throw Error.Argument(nameof(hostDocument), SRResource.HostDocumentIsNull);
+            }
+
+            _reference = new OpenApiReference()
+            {
+                Id = referenceId,
+                HostDocument = hostDocument,
+                Type = ReferenceType.Example,
+                ExternalResource = externalResource
+            };
+        }
+
+        /// <inheritdoc/>
+        public override string Description { get => _description ?? Target.Description; set => _description = value; }
```

Comment:
strings should be tested with isnullorempty instead of null coalesce to guard against empty strings when NRT is enabled.